### ISSUE TITLE
installing on OS X

### ISSUE THREAD

**Issue by laerm0** [riban-bw/vzeditor#10]
Hi. I'm trying to build for OS X (err, macOS now). The readme says it *should* work on OS X but I can't figure out how to build it. It appears the stumbling block is that `wxMidi` is not available for OS X. Homebrew has `portmidi` and `wxwidgets` but that's it. The readme also says source for `wxMidi` is included but it appears not to link when building the app.

FWIW the command I am using is `gcc -Iinclude `wx-config --cxxflags --libs` src/*.cpp`. Looks like it stops at 
```
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I am not at all experienced with C, though, and only made it this far with the help of a friend who is. Any guidance you might have is greatly appreciated. Thanks.

**Comment by laerm0** [comment]
Oh, huh, update already.

I got it built using this command: `clang++ -Iinclude 'wx-config --cxxflags'  'wx-config --libs' -lportmidi src/*.cpp -v -o vzeditor`

Good! But we have errors.

First, upon starting the program:

![screenshot_2020-01-06_00-11-37](https://user-images.githubusercontent.com/10780161/71796751-42210f80-3019-11ea-981b-e775f31ba215.png)

Then it looks like we have a resource path hard-coded…

![screenshot_2020-01-06_00-11-52](https://user-images.githubusercontent.com/10780161/71796763-506f2b80-3019-11ea-8baa-ed10f7962d9b.png)

But it might work? I'll report back in a few days when I know more. Thanks!

**Comment by riban-bw** [comment]
I can confirm the image assert messages and hardcoded keyboard image location. I see it on Ubuntu. My Windows build environment is broken - I have spent a few days trying to recover it but have given up and moved back to Ubuntu to test this. I will resolve those issues when I get a chance and look at your feedback about macOS. Thanks for the feedback - it is useful. I have no method of testing macOS builds.

**Comment by riban-bw** [comment]
I have updated master branch with fixes for the issues you highlighted. There was some quick and dirty code in there to get something to work with but I have tidied up the GUI code so it now uses a xpm for the keyboard display (compiled in rather than loaded from file at run time) and have fixed the (many) inappropriate growable element configurations.
Thanks for the build tip. I have tested this on Linux with g++ instead of clang and it kinda works but not completely. I will look at what magic Code::Blocks is doing for me and try to convert that into a Makefile.
